components/FiltersSection/__tests__/Brands.test.tsx

```diff
@@ -1,29 +1,26 @@
-import { GlobalContext } from '@/context/GlobalStore';
+import * as GlobalContext from '@/context/GlobalStore';
+import { mockBrands } from '@/utils/testData';
 import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { useState } from 'react';
 import { Brands } from '../brands';
 
 const user = userEvent.setup();
 
+jest
+  .spyOn(GlobalContext, 'useGlobalStore')
+  .mockReturnValue({
+    state: { brands: mockBrands, categories: [], cart: [] },
+    dispatch: jest.fn,
+  });
+
 const MockBrands = () => {
```

Comment:
Why are you creating a new component for the test here? render the actual component `Brands` with `handleChange` as a `jest.fn()` and expect the calls on that jest mock function.